### ISSUE TITLE
Dragging multiple nodes causes incorrect connectors position

### ISSUE THREAD

**Issue by Helen1987** [sporritt/jsPlumb#533]
I have several draggable nodes. Each node has children with source anchors. Those anchors are  connected with connections. When I drag several nodes at once, I see tangled connections as result.
I resolved it with `repaint` event after every drag, but it is a performance hit for 300 nodes.
Seems I found the reason for this bug. It was caused by [commit](https://github.com/sporritt/jsPlumb/commit/7bb246d690473a9a4606b3b008f40cd9903c3e6d).
The trouble is `options[stopEvent]` handler. It gets positions only ones:
`_currentInstance.getUIPosition(arguments);`
and applies it for all elements in selection:
`if (uip != null) _draw(_e[0], uip);`
As result, the first node is rendered correctly and other nodes anchors are moved to the place of the first node. As temporary solution I have replaced it with an old version: 
`_draw(_e[0], _e[1]);`
But I see that `getUIPosition` method is not straightforward. It takes `finalPos` or `pos` parameter to calculate the position. So, it relates to `ghostProxyOffsets` somehow. I have no idea what is  `ghostProxyOffsets`, so it can break something.

**Comment by sporritt** [comment]
do you have a jsfiddle demonstrating the issue?

**Comment by Helen1987** [comment]
I created [one](https://jsfiddle.net/Helen1987/ya5ukLqa/1/). Just click select button and drag any node. You will see that when you stop, connections are tangled.

**Comment by sporritt** [comment]
thanks

the curious thing is that if you go here

https://jsplumbtoolkit.com/demos/toolkit/flowchart-builder/index.html

and use the lasso to select a few nodes, you can drag them around just fine without this issue occurring at the end.  under the hood the toolkit uses the `addToDragSelection` method in the community edition.

**Comment by sporritt** [comment]
can you update it to use the non minified jsplumb. it's kind of difficult to debug with minified jsplumb.

**Comment by sporritt** [comment]
..forget that. i dug out my jsfiddle login and forked it.

**Comment by Helen1987** [comment]
Interesting... is there any specific actions after drag in toolkit? as I said, repaint in the end fix my troubles but it is a performance hit... But anyway the code I mentioned looks suspicious anyway.

**Comment by sporritt** [comment]
yes, after posting that i remembered that the Toolkit runs an update on each of the nodes in the drag, which ultimately causes a `revalidate` of the related dom element. This is analogous to what happens in this revision of your fiddle:

https://jsfiddle.net/sporritt/k4963L4r/3/

```
    jsPlumb.draggable(nodeElement, {
    	stop:function(p) {
      	p.selection.forEach(function(e) {
        	jsPlumb.revalidate(e[0]);
        })
      }
    });
```

but i do agree, the internal stop handler looks suspect. it has the selection to hand, and in each of them it has a position object, so in theory it should be using those.

**Comment by sporritt** [comment]
@Helen1987 there's a fix for this in 2.1.5, which is on a dev branch.

updated fiddle here:

https://jsfiddle.net/sporritt/k4963L4r/4/

**Comment by Helen1987** [comment]
Great! Thanks
